### PR TITLE
drivers: modem: gsm: Fix to COPS cmd parse info

### PR DESCRIPTION
**Pull request by icatorze**
drivers: modem: gsm: Quectel EC21 and BG9x act as a gsm modem without problem, but COPS commands can returns just one value with 'mode'.

Format and oper are not mandatory. This modification reads 'mode'.
Also a modification on modem_cmd_read_cops_cmd is needed.

### CODE REVIEW

**Review by mniestroj — APPROVED** [review]

File: drivers/modem/gsm_ppp.c
Comment (mniestroj) [inline drivers/modem/gsm_ppp.c:450]: One question, shoudn't it be `1U, 3U`, as previously `3U` was used?

Comment (icatorze) [reply drivers/modem/gsm_ppp.c:450]: Quectel documentation specifies 4 fields to COPS output, normally just 1 is used. But if in future use, like when custom commands list is used, we can get all used.
+COPS: <mode>[,<format>[,<oper>][,<Act>]]

**Review by rerickson1 — APPROVED** [review]

**Comment by icatorze** [comment]
did this workflow stuck?

**Comment by rerickson1** [comment]
> did this workflow stuck?

Try rebasing on main

**Comment by icatorze** [comment]
Sorry but: 4 workflows awaiting approval - First-time contributors need a maintainer to approve running workflows.

**Pull request merged**
